### ISSUE TITLE
Fixing checksum on ExHiROM games can take two iterations

### ISSUE THREAD

**Issue by Ramsis-SNES** [Optiroc/SuperFamicheck#3]
Hi again!

When you have an ExHiROM game where _the checksum bytes themselves_ are corrupt (more precisely, when `checksum != (checksum complement XOR $FFFF)`, it takes two iterations to actually fix the checksum.

To reproduce, with Tales of Phantasia as an example:

- Go to offset `$40FFDC`, and overwrite the checksum bytes (should be `A8 83 57 7C`) with, say, `12 34 56 78`
- Check and fix the ROM. SuperFamicheck will output the following:

 ```
superfamicheck tales.sfc -f
ROM info for file "tales.sfc"

  Header at   0x40ffb0
  Title       TALES OF PHANTASIA   
  Game code   ATVJ
  Maker code  AF
  Country     0x00 (Japan)
  Version     1.0

  ROM size    0x0d (8192KB, actual size 6144KB)
  RAM size    0x03 (8KB)
  Map mode    0x05 (Extended HiROM)
  Chipset     0x02 (ROM, RAM, Battery)
  Speed       120ns

  Checksum    0x7856
  Complement  0x3412

The following issues were found:
  Checksum/complement should be 0x7b6d/0x8492

Writing ROM image to file "tales.sfc"
  Fixed checksum
```
- The new checksum is obviously bad, too. However, running the command once more yields the correct result:

```
superfamicheck tales.sfc -f
ROM info for file "tales.sfc"

  Header at   0x40ffb0
  Title       TALES OF PHANTASIA   
  Game code   ATVJ
  Maker code  AF
  Country     0x00 (Japan)
  Version     1.0

  ROM size    0x0d (8192KB, actual size 6144KB)
  RAM size    0x03 (8KB)
  Map mode    0x05 (Extended HiROM)
  Chipset     0x02 (ROM, RAM, Battery)
  Speed       120ns

  Checksum    0x7b6d
  Complement  0x8492

The following issues were found:
  Checksum/complement should be 0x7c57/0x83a8

Writing ROM image to file "tales.sfc"
  Fixed checksum
```
This means that fixing the checksum within one iteration only works if the checksum is bad but its complement still equals `checksum XOR $FFFF`.

This issue only seems to be affecting ExHiROM games. I also tried it with Super Mario World (4 Mbit LoROM) and Chrono Trigger (32 Mbit HiROM), and it fixed the checksum in both cases immediately.

Thanks! :-)

**Comment by Optiroc** [comment]
I'm finally getting around to fix these issues. Doing this one tomorrow! ;)

**Comment by Optiroc** [comment]
Fixed in 5017dc7df5fd4d9bf3832bd50ed55a10b349cd35
